File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.js
function solution(triangle) {
    const n = triangle.length;
    const memo = new Array(n).fill(0);

    for (let i = n - 1; i >= 0; i--) {
        for (let j = 0; j <= i; j++) {
            if (i === n - 1) {
                memo[j] = triangle[i][j];
            } else {
                memo[j] = Math.max(memo[j], memo[j + 1]) + triangle[i][j];
            }
        }
    }

    return memo[0];
}

File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.py
def solution(triangle):
    
    height = len(triangle) - 1
    
    while height > 0:
        
        for i in range(len(triangle[height])-1):
            triangle[height-1][i] += max(triangle[height][i], triangle[height][i+1])
        height -= 1
        
    return triangle[0][0]